test/ctim/specs/fields_test.clj

```diff
@@ -0,0 +1,753 @@
+    (testing (pr-str key)
+         (are [expected value spec entity]
+             (= expected
+                (spec/valid? spec
+                             (assoc entity key value)))
+             false nil              plain-kw  plain-ex
+             false nil              new-kw    new-ex
+             false nil              stored-kw stored-ex
+             true  ""               plain-kw  plain-ex
+             true  ""               new-kw    new-ex
+             true  ""               stored-kw stored-ex
+             true  (rand-str 100)   plain-kw  plain-ex
+             true  (rand-str 100)   new-kw    new-ex
+             true  (rand-str 100)   stored-kw stored-ex
+             true  (rand-str 1000)  plain-kw  plain-ex
+             true  (rand-str 1000)  new-kw    new-ex
+             true  (rand-str 1000)  stored-kw stored-ex
+             true  (rand-str 5000)  plain-kw  plain-ex
+             true  (rand-str 5000)  new-kw    new-ex
+             true  (rand-str 5000)  stored-kw stored-ex
+             false (rand-str 5001)  plain-kw  plain-ex
+             false (rand-str 5001)  new-kw    new-ex
+             false (rand-str 5001)  stored-kw stored-ex
+             false (rand-str 10000) plain-kw  plain-ex
+             false (rand-str 10000) new-kw    new-ex
+             false (rand-str 10000) stored-kw stored-ex))))
+
+(defn test-long-string-in [key-path]
+  (let [plain-kw  (get-type-kw 'plain)
+        new-kw    (get-type-kw 'new)
+        stored-kw (get-type-kw 'stored)
+
+        plain-ex  (get-example 'plain)
+        new-ex    (get-example 'new)
+        stored-ex (get-example 'stored)]
+    (testing (pr-str key-path)
+      (are [expected value spec entity]
+          (= expected
+             (spec/valid? spec
+                          (assoc-in entity key-path value)))
+          false nil              plain-kw  plain-ex
+          false nil              new-kw    new-ex
+          false nil              stored-kw stored-ex
+          true  ""               plain-kw  plain-ex
+          true  ""               new-kw    new-ex
+          true  ""               stored-kw stored-ex
+          true  (rand-str 100)   plain-kw  plain-ex
+          true  (rand-str 100)   new-kw    new-ex
+          true  (rand-str 100)   stored-kw stored-ex
+          true  (rand-str 1000)  plain-kw  plain-ex
+          true  (rand-str 1000)  new-kw    new-ex
+          true  (rand-str 1000)  stored-kw stored-ex
+          true  (rand-str 5000)  plain-kw  plain-ex
+          true  (rand-str 5000)  new-kw    new-ex
+          true  (rand-str 5000)  stored-kw stored-ex
+          false (rand-str 5001)  plain-kw  plain-ex
+          false (rand-str 5001)  new-kw    new-ex
+          false (rand-str 5001)  stored-kw stored-ex
+          false (rand-str 10000) plain-kw  plain-ex
+          false (rand-str 10000) new-kw    new-ex
+          false (rand-str 10000) stored-kw stored-ex))))
+
+(deftest test-field-validators
```

Comment:
What is the purpose of this test ? 
To validate ShortString, MedString, LongString and Markdown schemas or to check that all string fields are correctly specified in entity schemas.

Reply:
Its meant to assert that the fields have the correct constraints.